src/olympia/shelves/serializers.py

```diff
@@ -27,3 +31,16 @@ def get_url(self, obj):
                     'user_pk': settings.TASK_USER_ID,
                     'collection_slug': obj.criteria})
         return url
+
+    def get_addons(self, obj):
+        if obj.endpoint == 'search':
+            endpoint = drf_reverse('v4:addon-search') + obj.criteria
+        elif obj.endpoint == 'collections':
+            endpoint = drf_reverse('v4:collection-addon-list', kwargs={
+                'user_pk': settings.TASK_USER_ID,
+                'collection_slug': obj.criteria})
+
+        url = settings.INTERNAL_SITE_URL + endpoint
+        response = requests.get(url)
```

Comment:
we don't want to use `requests` at all really- we want to call the APIView directly and get the results before it's been turned into a response using `AddonSearchView(request=request).data` - the tricky part is creating a fake request that at least has the correct GET parameters (i.e. the criteria) so it returns the correct data.  (Best case would be cloning the original `self.context.get('request')` and modifying it, somehow, but whatever works)

https://docs.djangoproject.com/en/2.2/ref/request-response/#httprequest-objects is the request object, though it's wrapped by Django Rest Framework in https://www.django-rest-framework.org/api-guide/requests/